### DISCUSSION TITLE
How to use uirevision attribute with ui.plotly.update

### DISCUSSION THREAD

**Discussion by Scheeer** [zauberzeug/nicegui#1386] [Q&A] [answered]
### Question

Plotly documentation outlines an option to keep user inputs while updating the data of a figure [(1)](https://plotly.com/javascript/uirevision/) 
```python

fig = {
    'data': [
        {
            'type': 'scatter',
            'name': 'Trace 1',
            'x': [1, 2, 3],
            'y': [0.5, 0.5, 0.5],
        },

    ],
    'layout': {
        'margin': {'l': 15, 'r': 0, 't': 0, 'b': 15},
        'uirevision':'true',
    },
}
plot = ui.plotly(fig).classes('w-full h-40')

def add_trace():
    fig["data"][0]["y"] = [random(), random(), random()]
    plot.update()
    
ui.button('Update', on_click=add_trace)

ui.run()

```
When using ui.plotly.update the user inputs are reset even though ui revision is defined in fig.
Is there a way to use uirevision in nicegui ?

**Comment by falkoschindler** [comment]
Hi @Scheeer,
It looks like this is currently not supported, mainly due to these lines: https://github.com/zauberzeug/nicegui/blob/02084d735a6f9da561bc3f55dade9440b9b00130/nicegui/elements/plotly.vue#L13-L15
Whenever the options are updated, a new plot is created.

I tried replacing it with
```py
      Plotly.react(this.$el.id, options.data, options.layout, options.config);
```
But now the "Update" button jumps around. This is the code I'm experimenting with:
```py
fig = {
    'data': [{'x': [1, 2, 3], 'y': [0.5, 0.5, 0.5]}],
    'layout': {'uirevision': True},
}
plot = ui.plotly(fig)

def add_trace():
    fig['data'][0]['y'] = [random(), random(), random()]
    plot.update()

ui.button('Update', on_click=add_trace)
```

Can anyone help to fix this layout problem?

**Reply by Scheeer** [reply]
interesting, on my old local version 1.2.17 replacing the update function works just fine without jumping buttons.

**Comment by falkoschindler** [comment, marked as answer]
Have a look at PR https://github.com/zauberzeug/nicegui/pull/1393 where I managed to successfully use `Plotly.react` (in most cases). This should solve your problem.

**Reply by Scheeer** [reply]
Great, thank you!
